### ISSUE TITLE
Unable to set script.allowed_contexts in elasticsearch.yml

### ISSUE THREAD

**Issue by sherry-ger** [elastic/elasticsearch#26651]
**Elasticsearch version** (`bin/elasticsearch --version`):
5.5.3

**Plugins installed**: []
x-pack

**JVM version** (`java -version`):
1.8.0_60

**Description of the problem including expected versus actual behavior**:

This is similar to the issue reported [here](https://github.com/elastic/elasticsearch/issues/24830).  

**Steps to reproduce**:

 1.  Add the following setting to elasticsearch.yml file per [documentation](https://www.elastic.co/guide/en/elasticsearch/reference/5.5/modules-scripting-security.html#allowed-script-contexts-setting)

```script.allowed_contexts: search, update```

 2. Start elasticsearch
 3. ES terminates with the following error:

```
[2017-09-14T11:16:54,820][WARN ][o.e.b.ElasticsearchUncaughtExceptionHandler] [cersei] uncaught exception in thread [main]
org.elasticsearch.bootstrap.StartupException: java.lang.IllegalArgumentException: unknown setting [script.allowed_contexts] please check that any required plugins are installed, or check the breaking changes documentation for removed settings
	at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:127) ~[elasticsearch-5.5.3.jar:5.5.3]
	at org.elasticsearch.bootstrap.Elasticsearch.execute(Elasticsearch.java:114) ~[elasticsearch-5.5.3.jar:5.5.3]
	at org.elasticsearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:67) ~[elasticsearch-5.5.3.jar:5.5.3]
	at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:122) ~[elasticsearch-5.5.3.jar:5.5.3]
	at org.elasticsearch.cli.Command.main(Command.java:88) ~[elasticsearch-5.5.3.jar:5.5.3]
	at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:91) ~[elasticsearch-5.5.3.jar:5.5.3]
	at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:84) ~[elasticsearch-5.5.3.jar:5.5.3]
Caused by: java.lang.IllegalArgumentException: unknown setting [script.allowed_contexts] please check that any required plugins are installed, or check the breaking changes documentation for removed settings
	at org.elasticsearch.common.settings.AbstractScopedSettings.validate(AbstractScopedSettings.java:293) ~[elasticsearch-5.5.3.jar:5.5.3]
	at org.elasticsearch.common.settings.AbstractScopedSettings.validate(AbstractScopedSettings.java:256) ~[elasticsearch-5.5.3.jar:5.5.3]
	at org.elasticsearch.common.settings.SettingsModule.<init>(SettingsModule.java:139) ~[elasticsearch-5.5.3.jar:5.5.3]
	at org.elasticsearch.node.Node.<init>(Node.java:343) ~[elasticsearch-5.5.3.jar:5.5.3]
	at org.elasticsearch.node.Node.<init>(Node.java:244) ~[elasticsearch-5.5.3.jar:5.5.3]
	at org.elasticsearch.bootstrap.Bootstrap$5.<init>(Bootstrap.java:232) ~[elasticsearch-5.5.3.jar:5.5.3]
	at org.elasticsearch.bootstrap.Bootstrap.setup(Bootstrap.java:232) ~[elasticsearch-5.5.3.jar:5.5.3]
	at org.elasticsearch.bootstrap.Bootstrap.init(Bootstrap.java:351) ~[elasticsearch-5.5.3.jar:5.5.3]
	at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:123) ~[elasticsearch-5.5.3.jar:5.5.3]
	... 6 more
```

**Comment by martijnvg** [comment]
@jdconrad It looks like the `ScriptModes.TYPES_ALLOWED_SETTING` and `ScriptModes.CONTEXTS_ALLOWED_SETTING` Setting instances aren't registered in `ClusterSettings.java`?

**Comment by jdconrad** [comment]
Guessing this got lost in the backport accidentally since the settings moved from the removed ScriptModes to ScriptService.

**Comment by rjernst** [comment]
Closed by #26703
